text/0521-find-by-identifier.md

```diff
@@ -0,0 +1,95 @@
+- Start Date: (fill me in with today's date, 2019-08-29)
+- Relevant Team(s): EmberData
+- RFC PR: https://github.com/emberjs/rfcs/pull/521
+- Tracking: (leave this empty)
+
+# [DATA] findRecord/peekRecord via Identifier
+
+## Summary
+
+Users should be able to peek or find records based on `lid`.
+
+## Motivation
+
+Apps and Addons making use of [Identifiers](https://github.com/emberjs/rfcs/pull/403)
+may wish to peek or find the record for a known `identifier`. This
+RFC would allow them to do so.
+
+## Detailed design
+
+A new method signature would be added to both `findRecord` and
+`peekRecord`. The existing signatures would be deprecated.
+
+Additionally `findRecord` would return a `Promise` resolving to a
```

Comment:
I'm definitely 👍 on moving away from `PromiseRecord`, but I'm a bit worried about having the same API emit different conceptual things (a stable object in the non-identifier signature, and a promise in the identifier based signature).

This isn't a super strong objection, and I'm totally willing to believe that this is just one of the ways we move away from the proxying but I wanted to notate the concern so we can discuss it specifically.

Reply:
Typescript allows us to specify which signature returns which return-type. Since users would be migrating to the new call signature in a location where they also handle the return, the would be able to account for this change all in one go. I think this is preferable to introducing a fully new API for querying for records at this time.

Reply:
Removed for deprecation at a later time